collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -211,9 +212,11 @@ public void initMap(MapFragment newMapFragment) {
         map.setFeatureClickListener(this::onFeatureClicked);
         updateInstanceGeometry();
 
-        if (selectedSubmissionId != -1) {
-            onFeatureClicked(selectedSubmissionId);
-        }
+        new Handler().postDelayed(() -> {
```

Comment:
Right, @seadowg? We should try not to have raw `Handler`s anymore so we can easily keep all those implementations consistent?

Reply:
It's not merged yet, so I think using `Handler` for the moment is best. I'm not quite sure why this code needs a `postDelayed` though?

Reply:
It was a fix for https://github.com/getodk/collect/issues/3828#issuecomment-628611800 but now I realized  it's still not perfect so I removed this trick at all. I'll try to find a better solution or leave it as-is since it's not a big problem.

Reply:
Ok I tried to figure out what's wrong but it's difficult. Since it's a small issue that affects only Mapbox and OSM let's create a separate issue to avoid blocking this pr:
![image](https://user-images.githubusercontent.com/3276264/85412710-2cbd8480-b56a-11ea-9ba9-7211d154debb.png)
 
@mmarciniak90 @kkrawczyk123 please keep it in mind.